icu4c/source/i18n/tibetancal.cpp

```diff
@@ -0,0 +1,573 @@
+  setTimeInMillis(getNow(), success); // Call this again now that the vtable is set up properly.
+}
+
+TibetanTsurphuCalendar::~TibetanTsurphuCalendar()
+{
+}
+
+const char *TibetanTsurphuCalendar::getType() const {
+    return "tibetan-tsurphu";
+}
+
+
+//-------------------------------------------------------------------------
+// Calculation utilities
+//-------------------------------------------------------------------------
+
+int32_t moonTab(int32_t i) {
+    // as in [Janson] (7.18)
+    static constexpr int32_t MOON_TAB[] = {
+        0,  5,  10,  15,  19,  22,  24,  25,  24,  22,  19,  15,  10,  5,  
+        0, -5, -10, -15, -19, -22, -24, -25, -24, -22, -19, -15, -10, -5 
+    }; 
+    return MOON_TAB[(i % 28 + 28) % 28];
+}
+
+int32_t sunTab(int32_t i) {
+    // as in [Janson] (7.21)
+    static constexpr int32_t SUN_TAB[] = {
+        0, 6, 10, 11, 10, 6, 0, -6, -10, -11, -10, - 6
+    }; 
+    return SUN_TAB[(i % 12 + 12) % 12];
+}
+
+// mean date
+constexpr double PHUGPA_M0 = 2015501 + 4783.0 / 5656; // as in [Janson] (7.4)
+constexpr double TSURPHU_M0 = 2353745 + 1795153.0 / 7635600; // as in [Janson] (A.3)
+// Mean length of the month, in the unit of days
+constexpr double M1 = 167025.0 / 5656; // as in [Janson] (7.2, 12.1, and A.1)
+// Mean length of the lunar day, in the unit of days
+constexpr double M2 = M1 / 30; // as in [Janson] (7.3, 12.2, Remark 14)
+
+//  Mean solar longitude at the epoch
+constexpr double PHUGPA_S0 = 743.0 / 804; // as in [Janson] (7.8)
+constexpr double TSURPHU_S0 = -5983.0 / 108540; // as in [Janson] (A.4)
+constexpr double S1 = 65.0 / 804; // as in [Janson] (7.6 and A.2)
+constexpr double S2 = S1 / 30; // as in [Janson] (7.7, and Remark 14)
+
+// Moon anomaly
+constexpr double PHUGPA_A0 = 475.0 / 3528; // as in [Janson] (7.14)
+constexpr double TSURPHU_A0 = 207.0 / 392; // as in [Janson] (A.5)
+constexpr double A1 = 253.0 / 3528; // as in [Janson] (7.12 and A.27)
+constexpr double A2 = 1.0 / 28; // as in [Janson] (7.13)
+
+constexpr int32_t PHUGPA_YEAR0 = 806;
+constexpr double PHUGPA_ALPHA = 1 + 827.0 / 1005; // as in [Janson] (C.30)
+
+constexpr int32_t TSURPHU_YEAR0 = 1732;
+constexpr double TSURPHU_ALPHA = 2 + 1903.0 / 18090; // as in [Janson] (C.44)
+
+
+/**
+ * Returns the julian date at the end of the lunar day (similar to the month count since the beginning of the epoch, but for days)
+ * It is calculated by first calculating a simpler mean date corresponding to the linear mean motion of the moon and then adjusting 
+ * it by the equations of the sun and moon, which are determined by the anomalies of the sun and moon together with tables.
+ * @param day the tibetan day
+ * @param monthCount month count since the begining of epoch
+ */
+int32_t TibetanCalendar::trueDate(int32_t day, int32_t monthCount) const {
+
+    double meanDate =  monthCount*M1 + day*M2 + PHUGPA_M0;
+    double moonAnomaly = 28.0*(monthCount*A1 + day*A2 + PHUGPA_A0);
+    int32_t mu = moonTab((int)ceil(moonAnomaly)); 
```

Comment:
please cast to the same type. Since you are using int32_t please cast to int32_t instead of int, and I think for new code we should use C++ style cast. Same as below.